src/app/chip-zcl-zpro/command-encoder/decoder.c

```diff
@@ -103,22 +103,18 @@ void printApsFrame(EmberApsFrame * frame)
            frame->groupId, frame->sequence, frame->radius);
 }
 
-void printCommandInfo(void * buffer)
+uint16_t extractMessage(uint8_t * buffer, uint16_t buffer_length, uint8_t ** msg)
 {
-    void * msg = NULL;
-    extractMessage(buffer, &msg);
-    uint8_t val = (uint8_t) * ((uint8_t *) (msg + 2));
-    printf("Encoded command id %d\n", val);
-}
-
-uint32_t extractMessage(void * buffer, void ** msg)
-{
-    uint32_t result = 0;
-    if (msg)
+    uint16_t result = 0;
+    if (msg && buffer_length >= 16)
     {
         // These are hard coded for now.
         *msg   = buffer + 13;
         result = 3;
```

Comment:
nit for future PRs: some constants names would be nice here. I am reading the code and don't actually undestand it - what is the offset 3 and why is size 3 the correct size.

Reply:
@bhaskar-apple ?

Reply:
Done